### DISCUSSION TITLE
onLoadingSlow will only be triggered when no cache.

### DISCUSSION THREAD

**Discussion by promer94** [vercel/swr#844] [Q&A]
i  would like to build a loading indicator for SWR.  
* if data fetching ends really fast ( 300ms <= ) ,  we will not show the spinner to users
* if data fetching ends slowly ( >= 800ms ), we will show the spinner to users.

isValidating is not very useful in this case. 

So i think i could make use of the onLoadingSlow event

```js
const [loading, setLoading] = useState(false)
const { data = [] } = useSWR('api', {
   onSuccess: () => {
     setLoading(false)
  },
  onLoadingSlow: () => {
      setLoading(false)
  },
  onError: () => {
   setLoading(false)
 },
 loadingTimeout: 800
})
return (
  <Spin loading={loading}>
    <Result data={data}>
  </Spin>
)
```

But i find that onLoadingSlow will only be triggered when no cache. This kind of behavior is different from other config callback options. 

Could we pass the current data to onLoadingSlow ```onLoadingSlow(data, key, config)``` instead of checking it.  So this callback function will always be triggered if it is a slow revalidation. I feel this is more flexible then current behavior

**Comment by xAl3xFx** [comment]
Hello, I have the same problem. Have you found any workaround with this?
